File: InterviewCampProblems/InterviewCampProblems.Code/Class9.cs
/*
Given an array of integers, rearrange the elements such that all zeros aremoved to the end of the array.
*/
using System;

namespace InterviewCampProblems.Code {
    public class Class9 {

        public static int[] MoveZerosToEnd (int[] arr) {

            if (arr == null || arr.Length == 0) {
                System.Console.WriteLine ("Null, Empty or Invalid array.");
                return arr;
            }

            int boundary = arr.Length - 1;
            for (int i = arr.Length - 1; i >= 0; i--) {
                if (arr[i] == 0) {
                    Swap (arr, i, boundary);
                    boundary--;
                }
            }
            return arr;
        }

        private static void Swap (int[] arr, int i, int boundary) {
            int temp = arr[i];
            arr[i] = arr[boundary];
            arr[boundary] = temp;
        }
    }
}

File: Java/InterviewCamp/src/main/java/Class9.java
//Given an array of integers, rearrange the elements such that all zeros are moved to the end of the array.

public class Class9 {

    public static int[] RearrangeArray(int[] arr)
    {
        if(arr == null || arr.length == 0) {
            return null;
        }

        int boundary =arr.length-1;
        for (int i = arr.length-1; i >= 0; i--) {

            if(arr[i] == 0)
            {
                swap(arr, i, boundary);
                boundary -= 1;
            }
        }
        return  arr;
    }

    private static void swap(int[] arr, int i, int boundary) {
        int temp = arr[i];
        arr[i] = arr[boundary];
        arr[boundary] = temp;
    }
}
